Bottom: 6f2a456960849d37dc43a77ea476a67b658609e9
Top:    7d477598a3023f00e634ca790d8a69d172b320c3
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-29 19:39:34 +0100

hid/gtk: Extend layer selector to allow editing layer names in situ


---

diff --git a/src/hid/gtk/ghid-layer-selector.c b/src/hid/gtk/ghid-layer-selector.c
index 8c7a713..6245b5c 100644
--- a/src/hid/gtk/ghid-layer-selector.c
+++ b/src/hid/gtk/ghid-layer-selector.c
@@ -302,6 +302,7 @@ ghid_layer_selector_init (GHidLayerSelector *ls)
 
   renderer1 = ghid_cell_renderer_visibility_new ();
   renderer2 = gtk_cell_renderer_text_new ();
+  g_object_set (renderer2, "editable-set", TRUE, NULL);
 
   opacity_col = gtk_tree_view_column_new_with_attributes ("", renderer1,
                                                           "active", VISIBLE_COL,
@@ -309,6 +310,7 @@ ghid_layer_selector_init (GHidLayerSelector *ls)
   name_col = gtk_tree_view_column_new_with_attributes ("", renderer2,
                                                        "text", TEXT_COL,
                                                        "font", FONT_COL,
+                                                       "editable", EDITABLE_COL,
                                                        NULL);
 
   ls = g_object_new (GHID_LAYER_SELECTOR_TYPE, NULL);
